### PR TITLE
Rhmap 11889 mbaas mongo conn

### PR DESCRIPTION
**Pull request by wei-lee**
## Motive

Setup the shared mongo db connections once for fh-forms so that it will reuse the existing mongodb connections

## Changes

Make sure fh-forms will setup the shared mongo db connections on fh-mbaas startup.

## JIRA

https://issues.jboss.org/browse/RHMAP-11889

This is related to PR https://github.com/feedhenry/fh-forms/pull/7

### CODE REVIEW

File: fh-mbaas.js [hunk 1]
```diff
@@ -121,11 +121,17 @@ function createAndSetLogger() {
 
   //Setting logger for fh-forms
   forms.init(logger);
-
-  //Setting global forms config
-  logger.debug("minsPerBackOffIndex", fhconfig.int('fhmbaas.dsMinsPerBackOffIndex'));
-  forms.core.setConfig({
-    minsPerBackOffIndex: fhconfig.int('fhmbaas.dsMinsPerBackOffIndex')
+  forms.setupSharedMongoConnections(logger, fhconfig.mongoConnectionString(), fhconfig.getConfig().rawConfig, function(err, sharedConnections) {
```
Comment (nialldonnellyfh) [inline fh-mbaas.js]: I would be very hesitant to give the fh-forms module access to this administrative account.

I would advise setting up a user with access all database permissions but without "admin" permissions. E.g dropping databases etc.

Comment (wei-lee) [inline fh-mbaas.js]: yes, that makes sense. Will create a new user for that

**Comment by wei-lee** [comment]
Retest this please

File: lib/util/mongo.js [hunk 2]
```diff
@@ -70,5 +73,62 @@ function dropDb(config, dbUser, dbName, cb) {
...
+ * @param {string} targetDbUrl the database url that the user should be created in
+ * @param {object} userParams user info
+ * @param {string} userParams.username the username
+ * @param {string} userParams.password the user password
+ * @param {array} userParams.roles the roles of the user. See mongodb user privilege doc: https://docs.mongodb.com/v2.4/reference/privilege-documents/
+ * @param {function} cb callback function
+ */
+function createDbUser(targetDbUrl, userParams, cb) {
+  logger.debug({"targetDbUrl": targetDbUrl, "user": userParams}, "createing new db user");
+  MongoClient.connect(targetDbUrl, function(err, targetDb) {
```
Comment (nialldonnellyfh) [inline lib/util/mongo.js]: Small thing, might want to use async.

File: lib/util/mongo.js [hunk 3]
```diff
@@ -70,5 +73,62 @@ function dropDb(config, dbUser, dbName, cb) {
...
+/**
+ * Create a new user in the target database with the given user info only if the user does not eixst.
+ * @param {string} targetDbUrl the database url that the user should be created in
+ * @param {object} userParams user info
+ * @param {string} userParams.username the username
+ * @param {string} userParams.password the user password
+ * @param {array} userParams.roles the roles of the user. See mongodb user privilege doc: https://docs.mongodb.com/v2.4/reference/privilege-documents/
+ * @param {function} cb callback function
+ */
+function createDbUser(targetDbUrl, userParams, cb) {
```
Comment (nialldonnellyfh) [inline lib/util/mongo.js]: This file is getting a bit overloaded, might want to create a new file.

**Pull request merged**